dbms/src/Common/Exception.h

```diff
@@ -172,28 +162,123 @@ std::enable_if_t<std::is_pointer_v<T>, T> exception_cast(std::exception_ptr e)
 
 namespace exception_details
 {
+inline std::string generateFormattedMessage(const char * condition)
+{
+    return fmt::format("Assert {} fail!", condition);
+}
+
 template <typename T, typename... Args>
-inline std::string generateLogMessage(const char * condition, T && fmt_str, Args &&... args)
+inline std::string generateFormattedMessage(const char * condition, T && fmt_str, Args &&... args)
+{
+    return FmtBuffer().fmtAppend("Assert {} fail! ", condition).fmtAppend(fmt_str, std::forward<Args>(args)...).toString();
+}
+
+template <typename... Args>
+inline Poco::Message generateLogMessage(const std::string & logger_name, const char * filename, int lineno, const char * condition, Args &&... args)
+{
+    return Poco::Message(
+        logger_name,
+        generateFormattedMessage(condition, std::forward<Args>(args)...),
+        Poco::Message::PRIO_FATAL,
+        filename,
+        lineno);
+}
+
+Poco::Logger * getDefaultFatalLogger();
```

Comment:
How about returning `LoggerPtr` then 
```cpp
inline void log(const char * filename, int lineno, const char * condition, Poco::Logger * logger)
```
and 
```cpp
inline void log(const char * filename, int lineno, const char * condition, Poco::Logger * logger, const char * fmt_str, Args &&... args)
```
can be removed.

Reply:
hmmm ... looks like we don't need to support `Poco::Loger *` at the first revision?

Reply:
I guess so...

Reply:
done